selfservice/strategy/oidc/provider_apple.go

```diff
@@ -146,3 +147,21 @@ func decodeQuery(query url.Values, claims *Claims) {
 		}
 	}
 }
+
+var _ IDTokenVerifier = new(ProviderApple)
+
+func (a *ProviderApple) Verify(ctx context.Context, rawIDToken string) (*Claims, error) {
+	keySet := oidc.NewRemoteKeySet(ctx, "https://appleid.apple.com/auth/keys")
+	verifier := oidc.NewVerifier("https://appleid.apple.com", keySet, &oidc.Config{
+		ClientID: a.config.ClientID,
```

Comment:
This is the only reason you'd need a configured apple client in the OIDC config, as it checks whether the IDToken was issued to the same `client_id`. As mentioned above, this means that for development using expo, you'd need to set it to `host.exp.Exponent`. 

I am not 100% sure this is secure enough, but since we verify that the token was issued by Apple, it _should_ be fine™️.

In production apps, the token would be issued to `sh.ory.iosapp` on the device, and the Ory project/configuration would need a client with the same `client_id`. `apple_team_id`, `apple_private_key_id` an `apple_private_key` would be ignored in this flow.